src/util/artifacts.js

```diff
@@ -1,25 +1,61 @@
 /* @flow */
 import {fs} from 'mz';
+import mkdirp from 'mkdirp';
+import promisify from 'es6-promisify';
 
 import {UsageError, isErrorWithCode} from '../errors';
 import {createLogger} from './logger';
 
 const log = createLogger(__filename);
+const defaultAsyncMkdirp = promisify(mkdirp);
 
+type PrepareArtifactsDirOptions = {
+  asyncMkdirp?: typeof defaultAsyncMkdirp,
+}
 
 export async function prepareArtifactsDir(
-  artifactsDir: string
+  artifactsDir: string,
+  {asyncMkdirp = defaultAsyncMkdirp}: PrepareArtifactsDirOptions,
```

Comment:
This looks good but in my example I forgot to say that this should be optional. You can tell that it needs to be optional because it has a default value. The definition to make it optional would look like this. The syntax is a bit weird but what it's doing is initializing the options parameter to an empty object.
````js
export async function prepareArtifactsDir(
  artifactsDir: string,
  {asyncMkdirp = defaultAsyncMkdirp}: PrepareArtifactsDirOptions = {},
): Promise<string> {
  ...
}
````
After that, all the tests can call it like before:
````js
return prepareArtifactsDir(tmpPath)
  ..
````